merino-suggest/src/lib.rs

```diff
@@ -341,6 +350,151 @@ pub enum LanguageIdentifier {
     Wildcard,
 }
 
+/// The form factor of the device that sent a given suggestion request.
+#[derive(Clone, Debug, Hash, PartialEq, Serialize)]
```

Comment:
Let's move these new structs to a separate file, this module's getting pretty large. I'm big about doing this up front rather than later because we'll be less likely to run into merge conflicts with each other in the future, especially during a crunch time (which hopefully will not occur in the first place :smile: ).

Plus it's always more difficult to do later for the same reason as the merge conflicts.

Reply:
not to muddy things, but kinda/sorta wonder if this might be a good candidate for `common`, since it's basically the same code being used in both  Merino and Contile.

Counter argument is that this code is also very specific to Contile and Merino and it's fine having the duplication since we're "fuzzing" things.

Reply:
Let's land this PR approximately as is. I agree that adding something like this to common would be a good idea in the future.